### ISSUE TITLE
MA0075 flags custom wrapper forwarding to `string.Create(CultureInfo.InvariantCulture, ref DefaultInterpolatedStringHandler)`

### ISSUE THREAD

**Issue by Piedone** [meziantou/Meziantou.Analyzer#1316]
### Version of the Meziantou.Analyzer NuGet package

3.0.173

### Rule Identifier

MA0075

### Target Framework

net10.0

### C# Language version

C# 14.0

### .NET SDK version

10.0.400

### Description

MA0075 ("Do not use implicit culture-sensitive ToString") is reported on calls to an extension method that takes a `DefaultInterpolatedStringHandler` parameter and forwards it to `string.Create(CultureInfo.InvariantCulture, ref value)` internally. This is the exact same pattern the analyzer already treats as safe when `string.Create(IFormatProvider, DefaultInterpolatedStringHandler)` is called directly.

Because the analyzer only looks at the call site and doesn't see that the callee forwards to the safe overload, it flags every call to the wrapper, even though (from the call site's perspective) the intent is identical to calling `string.Create(CultureInfo.InvariantCulture, $"...")` directly.

### Reproduction Steps

Minimal code:

```c#
using System;
using System.Globalization;
using System.Runtime.CompilerServices;

public static class StringHelper
{
    // MA0075 is reported at every call site below, even though this forwards to the same
    // string.Create(IFormatProvider, DefaultInterpolatedStringHandler) overload the analyzer
    // already treats as culture-safe when called directly.
    public static string CreateInvariant(this DefaultInterpolatedStringHandler value) =>
        string.Create(CultureInfo.InvariantCulture, ref value);
}

public static class Program
{
    public static string FormatDate(DateTime value) =>
        StringHelper.CreateInvariant($"Date: {value:yyyy-MM-dd}"); // MA0075 fires here

    public static string FormatDateDirect(DateTime value) =>
        string.Create(CultureInfo.InvariantCulture, $"Date: {value:yyyy-MM-dd}"); // no warning
}
```

### Other information

Found while investigating new warnings introduced after bumping Meziantou.Analyzer from 3.0.141 to 3.0.159+ in https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions.

**Comment by meziantou** [comment]
The issue is that DefaultInterpolatedStringHandler is considered as culture-sensitive. So, calling `StringHelper.CreateInvariant` reports the diagnostic.
A solution would be to add a new attribute to annotate the parameter, so call-site will be aware of it.

**Comment by Piedone** [comment]
Thanks for your quick reply!

To confirm, is this isn't a wrong usage of `DefaultInterpolatedStringHandler` in your opinion, right? Because if it isn't, then I have no better idea than an attribute (i.e. some way to mark the parameter for the analyzer to just leave it), apart from more involved analysis that would go beyond normal static analysis.

**Comment by meziantou** [comment]
In fact, I'm not sure your code is doing what you expect. The `DefaultInterpolatedStringHandler` is created in `FormatDate` without the knowledge of the CultureInfo. So, CreateInvariant may use the culture to late.
I should write a test to confirm it.

Anyway the parameter would be useful in some cases, but maybe not this one.

**Comment by meziantou** [comment]
`CreateInvariant` doesn't make anything invariant — it formats with the current culture.

The interpolated string is formatted *before* `CreateInvariant` is entered. At the call site the compiler builds the `DefaultInterpolatedStringHandler` with the `(int literalLength, int formattedCount)` constructor — no `IFormatProvider`, because the parameter has no `[InterpolatedStringHandlerArgument]` attribute. So `AppendFormatted(value, "yyyy-MM-dd")` runs against `CultureInfo.CurrentCulture` right there.

By the time `string.Create(CultureInfo.InvariantCulture, ref value)` receives the handler, the text is already built. The provider is ignored; `string.Create` just calls `ToStringAndClear()` on it.

It looks correct in most testing because `-` in `yyyy-MM-dd` is a literal, not the culture's date separator (`/` is), so every Latin-calendar culture gives the same output. Non-Gregorian calendars don't:

    [Fact]
    public void FormatDate_IsNotInvariant()
    {
        // th-TH defaults to the Thai Buddhist calendar
        CultureInfo.CurrentCulture = CultureInfo.GetCultureInfo("th-TH");

        Assert.Equal("Date: 2026-08-20", StringHelper.FormatDate(new DateTime(2026, 8, 20)));
        // Assert.Equal() Failure
        // Expected: Date: 2026-08-20
        // Actual:   Date: 2569-08-20
    }

Same under `ar-SA` (Umm al-Qura): `Date: 1448-03-07`.

The helper can't be fixed while keeping its shape — the handler has no provider to receive, and an extension method has nowhere to put the `IFormatProvider` that `[InterpolatedStringHandlerArgument]` needs to point at. The call has to construct the handler itself:

    public static string FormatDate(DateTime value) =>
        string.Create(CultureInfo.InvariantCulture, $"Date: {value:yyyy-MM-dd}");

That returns `Date: 2026-08-20` under invariant, en-US, fr-FR, th-TH and ar-SA alike.

-- Claude (with an actual test to validate the behavior)

**Comment by Piedone** [comment]
I see, thank you for looking into this. My use case is invalid, then. Please close this if you think that should be the end of it and the attribute isn't even needed.
